src/components/NavBar/SearchBarDropdown.tsx

```diff
@@ -212,162 +258,114 @@ export const SearchBarDropdown = ({
-                  suggestion_type: NavBarSearchTypes.COLLECTION_TRENDING,
-                  ...eventProperties,
-                }}
-                header={<Trans>Popular NFT collections</Trans>}
-                headerIcon={<TrendingArrow />}
-                isLoading={trendingCollectionsAreLoading}
-              />
-            )}
-          </Column>
-        )
-
-      setResultsState(currentState)
-    }
-  }, [
-    isLoading,
-    tokens,
-    collections,
-    formattedTrendingCollections,
-    trendingTokens,
-    trendingTokenData,
-    hoveredIndex,
-    toggleOpen,
-    shortenedHistory,
-    hasInput,
-    isNFTPage,
-    isTokenPage,
-    showCollectionsFirst,
-    queryText,
-    totalSuggestions,
-    trace,
-    searchHistory,
-    trendingCollectionsAreLoading,
-  ])
+  const tokenSearchResults =
+    tokens.length > 0 ? (
+      <SearchBarDropdownSection
+        hoveredIndex={hoveredIndex}
+        startingIndex={showCollectionsFirst ? collections.length : 0}
+        setHoveredIndex={setHoveredIndex}
+        toggleOpen={toggleOpen}
+        suggestions={tokens}
+        eventProperties={{
+          suggestion_type: NavBarSearchTypes.TOKEN_SUGGESTION,
+          ...eventProperties,
+        }}
+        header={<Trans>Tokens</Trans>}
+      />
+    ) : (
+      <Box className={styles.notFoundContainer}>
+        <Trans>No tokens found.</Trans>
+      </Box>
+    )
 
-  const showBNBComingSoonBadge = chainId === SupportedChainId.BNB && !isLoading
+  const collectionSearchResults =
+    collections.length > 0 ? (
+      <SearchBarDropdownSection
+        hoveredIndex={hoveredIndex}
+        startingIndex={showCollectionsFirst ? 0 : tokens.length}
+        setHoveredIndex={setHoveredIndex}
+        toggleOpen={toggleOpen}
+        suggestions={collections}
+        eventProperties={{
+          suggestion_type: NavBarSearchTypes.COLLECTION_SUGGESTION,
+          ...eventProperties,
+        }}
+        header={<Trans>NFT Collections</Trans>}
+      />
+    ) : (
+      <Box className={styles.notFoundContainer}>No NFT collections found.</Box>
+    )
 
   return (
-    <Column overflow="hidden" className={clsx(styles.searchBarDropdownNft, styles.searchBarScrollable)}>
-      <Box opacity={isLoading ? '0.3' : '1'} transition="125">
-        {resultsState}
-        {showBNBComingSoonBadge && (
-          <BNBComingSoonBadge>
-            <BNBLogo src={BnbLogoURI} />
-            <ThemedText.BodySmall color="textSecondary" fontSize="14px" fontWeight="400" lineHeight="20px">
-              <Trans>Coming soon: search and explore tokens on BNB Chain</Trans>
-            </ThemedText.BodySmall>
-          </BNBComingSoonBadge>
-        )}
-      </Box>
-    </Column>
+    <>
+      {hasInput ? (
+        // Empty or Up to 8 combined tokens and nfts
+        <Column gap="20">
+          {showCollectionsFirst ? (
+            <>
+              {collectionSearchResults}
+              {tokenSearchResults}
```

Comment:
these are conditionally rendered, so should we just inline the jsx? there may be a perf hit from putting them in variables before we know they're needed

Reply:
i think actually the perf hit is only in an extra createElement() call with an object, at least i think react only does the sub-tree stuff based on what's actually returned not at the call-site